src/Planner/PlannerActionsVisitor.cpp

```diff
@@ -182,6 +182,30 @@ class ActionNodeNameHelper
                     break;
                 }
 
+                /// Function "tuple" which generates named tuple should use argument aliases to construct its name.
+                if (function_node.getFunctionName() == "tuple")
+                {
+                    if (const DataTypeTuple * type_tuple = typeid_cast<const DataTypeTuple *>(function_node.getResultType().get()))
+                    {
+                        if (type_tuple->haveExplicitNames())
```

Comment:
This will likely not work in the distributed case because the query on the follower node won't have `AS col_a` and `AS col_b` for the `tuple` function argument. I think a proper fix would also include creating a cast operation.

Reply:
Also, I think that such code should check `enable_named_columns_in_function_tuple` setting.

Reply:
> This will likely not work in the distributed case because the query on the follower node won't have `AS col_a` and `AS col_b` for the `tuple` function argument. I think a proper fix would also include creating a cast operation.

It works because we do `addConvertingActions` in ReadFromRemote step. I've added a test case to verify this.

> Also, I think that such code should check enable_named_columns_in_function_tuple setting.

Sure.